podcasts/PaidFeature.swift

```diff
@@ -0,0 +1,84 @@
+import Combine
+import PocketCastsServer
+import SwiftUI
+
+// MARK: - Features
+
+/// To add a new feature:
+///     1. Add a new static var for the feature name and tier it should be unlocked with
+///
+///     Template:
+///         static var <#FeatureName#>: PaidFeature = .init(tier: <#Tier#>)
+///
+///     3. Check the unlock state using `PaidFeature.hello.isUnlocked`
+///
+extension PaidFeature {
+    static var bookmarks: PaidFeature = .init(tier: .patron)
+}
+
+/// A `PaidFeature` represents a feature that is unlocked with a subscription tier, and is considered to be unlocked if the tier
+/// is equal to or higher than the set `tier` value.
+///
+/// The unlock state is self managed by listening for relevant notifications and updating the `isUnlocked` property accordingly.
+///
+/// Since this is a subclass of `ObservableObject` it can easily be used in SwiftUI. Outside SwiftUI, you can use the `objectWillChange`
+/// publisher to be notified about changes.
+///
+/// And while the class is an `ObservableObject` it doesn't use any `@Published` properties and instead manually triggers `objectWillChange`.
+/// This is done to ensure future compatibility of any property changes by not allowing listeners to directly access a published properties publisher.
+///
+class PaidFeature: ObservableObject {
+    /// Whether the feature is unlocked for the active subscription tier
+    var isUnlocked: Bool {
+        subscriptionHelper.activeTier >= tier
+    }
+
+    /// The minimum subscription level required to unlock this feature
+    let tier: SubscriptionTier
+
+    /// The static class to use to check for the active subscription.
+    private let subscriptionHelper: SubscriptionHelper.Type
+
+    private var cancellables = Set<AnyCancellable>()
+
+    /// Creates a new paid feature with the minimum `unlockTier`.
+    init(tier: SubscriptionTier, subscriptionHelper: SubscriptionHelper.Type = SubscriptionHelper.self) {
+        self.tier = tier
+        self.subscriptionHelper = subscriptionHelper
+
+        addListeners()
+    }
+
+    /// Listen for changes and update the internal state
+    private func addListeners() {
+        Publishers.Merge(
+            ServerNotifications.iapPurchaseCompleted.publisher(),
+            ServerNotifications.subscriptionStatusChanged.publisher()
+        )
+        .receive(on: DispatchQueue.main)
+        .sink { [weak self] _ in
+            self?.objectWillChange.send()
+        }
+        .store(in: &cancellables)
+    }
+}
+
+// MARK: - Helpers
```

Comment:
I added these because I found myself repeating the same logic in a few places so it made sense to make it reusable.